documents/framework.md

```diff
@@ -204,4 +204,75 @@ CELIX_AUTO_START_3=FooBundle.zip
+.cache/bundle1/version1.0
+.cache/bundle1/version1.0/libshelld.so.2
+.cache/bundle1/version1.0/libshell.so.2
+.cache/bundle1/version1.0/META-INF
+.cache/bundle1/version1.0/META-INF/MANIFEST.MF
+.cache/bundle2
+.cache/bundle2/bundle_state.properties
+.cache/bundle2/storage
+.cache/bundle2/version1.0
+.cache/bundle2/version1.0/libshell_tuid.so.1
+.cache/bundle2/version1.0/META-INF
+.cache/bundle2/version1.0/META-INF/MANIFEST.MF
+```
+
+The entry `.cache/bundle1/version1.0` is the resource bundle cache and the entry `.cache/bundle1/storage` is the 
+persistent storage bundle cache for the `Celix::shell` bundle.
+
+## Framework configuration options
+The Apache Celix framework can be configured using framework properties. 
+
+The framework properties can be provided in the following ways:
+ - Using the Apache Celix launcher with a "config.properties" file.
+ - Creating a framework using the framework factory and providing a celix_properties_t*.
+ - Setting environment variables with the prefix "CELIX_". 
+
+Note that the config properties and environment variables are only read once when the framework is created. 
+So changing the environment variables after the framework is created will not have any effect.
+
+The following framework properties are supported:
+
+| Framework Property                                  | Default Value | Description                                                                                                                               |
+|-----------------------------------------------------|---------------|-------------------------------------------------------------------------------------------------------------------------------------------|
+| CELIX_FRAMEWORK_CACHE_DIR                           | ".cache"      | The directory where the Apache Celix framework will store its data.                                                                       |
+| CELIX_FRAMEWORK_CACHE_USE_TMP_DIR                   | "false"       | If true, the Apache Celix framework will use the system temp directory for the cache directory.                                           |
+| CELIX_FRAMEWORK_CLEAN_CACHE_DIR_ON_CREATE           | "false"       | If true, the Apache Celix framework will clean the cache directory on create.                                                             |
+| CELIX_FRAMEWORK_FRAMEWORK_UUID                      | ""            | The UUID of the Apache Celix framework. If not set, a random UUID will be generated.                                                      |
+| CELIX_BUNDLES_PATH                                  | "bundles"     | The directories where the Apache Celix framework will search for bundles. Multiple directories can be provided separated by a colon.      |
+| CELIX_LOAD_BUNDLES_WITH_NODELETE                    | "false"       | If true, the Apache Celix framework will not delete bundles when they are uninstalled. Note for cmake build type Debug, default is "true" |
```

Comment:
This influences dlopen behavior delete bundles reads as removal of files. Propose rephrase

Reply:
rephrased. But note that the dlopen flags is named RTLD_NODELETE as a false will result in a "delete" behaviour.